src/plugins/WebOfTrust/util/SynchronousDelayedBackgroundJob.java

```diff
@@ -0,0 +1,231 @@
+    }
+
+    /**
+     * Triggers the execution of the job after either aggregating triggers for at most the given
+     * delay, or as soon as the currently running job finishes, whichever comes last. Negative
+     * delays are treated as to zero delays.
+     * If this background job was terminating or has terminated prior to the invocation of this
+     * method, this method has no effect and returns immediately.
+     * This method waits for the first job to start after the trigger to finish before returning,
+     * except when this method is called from the background job itself, in that case this method
+     * does not wait for the job to finish (otherwise a deadlock would be the result).
+     * If the calling thread is interrupted while waiting for the job to finish execution, this
+     * method will return as soon as possible, without waiting for the job to finish, and sets the
+     * thread interruption flag.
+     * @param delay the maximum trigger aggregation delay in milliseconds
+     */
+    @Override
+    public synchronized void triggerExecution(long delay) {
+        if (isTerminating) {
+            return;
+        }
+        triggerExecutionAsynchronously(delay);
+        // When called from the job thread, don't wait for the job to finish: this would cause a
+        // deadlock.
+        if (runningJobThread == Thread.currentThread()) {
+            return;
+        }
+        // Wait for the relevant execution to complete.
+        long waitForRunCount;
+        if (runningJobThread == null) {
+            // Wait for next run to finish
+            waitForRunCount = runCount + 1;
+        } else {
+            // Wait for next run after current run to finish
+            waitForRunCount = runCount + 2;
+        }
+        while (runCount < waitForRunCount) {
+            try {
+                wait();
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+                return;
+            }
+        }
+    }
+
+    /**
+     * Implementation of {@link #triggerExecution(long)} that does everything expected from that
+     * method, except it does not check for termination (the caller is assumed to have done this)
+     * and does not wait for the execution to complete.
+     * @see #triggerExecution(long)
+     * Caller must ensure synchronization on {@code this}.
+     */
+    private void triggerExecutionAsynchronously(long delay) {
+        if (delay < 0) {
+            delay = 0;
+        }
+        long newDeadline = System.currentTimeMillis() + delay;
+        if (newDeadline < nextRunDeadline) {
+            nextRunDeadline = newDeadline;
+            if (!nextRunScheduled) {
+                // Create a new waiting job thread using the given delay.
+                nextRunScheduled = true;
+                startNewJobThread();
+            } else {
+                // Notify waiting job thread that it should start earlier.
+                notifyAll();
```

Comment:
Not fixing this as part of my own fixes since it would be much work and the better solution would be to replace the whole of the threading in the class with TickerDelayedBackgroundJob as discussed elsewhere.

-> Done.